virtme/guest/virtme-sshd-script

```diff
@@ -46,4 +46,10 @@ for key in "${CACHE_DIR}"/etc/ssh/ssh_host_*_key; do
     ARGS+=(-h "${key}")
 done
 
-/usr/sbin/sshd "${ARGS[@]}"
+if [ ! -c /dev/vsock ]; then
+    /usr/sbin/sshd "${ARGS[@]}"
+else
+    # 4294967295 == U32_MAX == -1
+    declare -r VMADDR_CID_ANY=4294967295
+    systemd-socket-activate --accept --listen="vsock:${VMADDR_CID_ANY}:22" --inetd -- /usr/sbin/sshd -i "${ARGS[@]}" &
```

Comment:
This part is still blocking the init for me. Maybe we can do something like this:
```
{
    if [ ! -c /dev/vsock ]; then
        /usr/sbin/sshd "${ARGS[@]}"
    else
        # 4294967295 == U32_MAX == -1
        declare -r VMADDR_CID_ANY=4294967295
        systemd-socket-activate --accept --listen="vsock:${VMADDR_CID_ANY}:22" --inetd -- /usr/sbin/sshd -i "${ARGS[@]}"
    fi
} </dev/null &>/dev/null &

Reply:
Hmm do you use the Bash version of `init` or the Rust? I see there are differences in regard to handling stdout, but that's a separate issue and should be fixed in the init files.

Does it behave different if you use `--ssh --force-tcp`?

Reply:
I'm using virtme-ng-init (the Rust init). If I force the bash init, it works (`--no-virtme-ng-init`). But the stdout is still messed up (extra newline, etc.).

Reply:
The thing with the output is a difference between the Rust and the Bash implementation - I can create another commit for fixing this... (at least partly)

In Rust:
stdin is set to /dev/null and only stderr (not stdout) is logged in case of success. The buffering is also different... e.g. Stdio::piped uses buffering and so on... [1]

In Bash:
Default settings for buffering is used and no redirection of stdout/stderr/stdin [2]

[1] https://github.com/arighi/virtme-ng/blob/main/virtme_ng_init/src/utils.rs#L147-L170
[2] https://github.com/arighi/virtme-ng/blob/main/virtme/guest/virtme-init#L268

Reply:
I've changed at least the redirection of STDIN (/dev/null) in a new commit:
https://github.com/arighi/virtme-ng/pull/261/commits/4ac06123d3768ea4e0a9552e647e548e8dc42341

For the "daemonization" of the `systemd-socket-activate ...` construct we now use `nohup` + stdout/stderr redirection.